src/spetlr/deltaspec/DeltaTableSpecDifference.py

```
@@ -0,0 +1,356 @@
+
+        if not self.location_match():
+            return False
+
+        # at this point we know that at least the name and location match
+        if self.schema_match():
+            return True
+
+        # so the schemas are different.
+        # The question remains if they are similar enough to allow reading.
+        return self.schema_is_selectable()
+
+    def schema_is_selectable(self):
+        if self.base is None:
+            return False
+        # if differences are only in nuallability and comment,
+        # we have no problem. This is handled implicitly.
+        # If the specified schema is a subset of the storage schema,
+        # then we can read without issue.
+
+        # TODO: For now we do not support reading a
+        #           subset of struct fields in the case of struct columns.
+        #       Each top level column has to match in type.
+
+        # Check if the specified schema is a subset of the disk schema.
+        storage_schema_dict: Dict[str, StructField] = {
+            field.name: field for field in self.base.schema.fields
+        }
+        for field in self.target.schema.fields:
+            # every field has to be in the storage, or we can stop
+            if field.name not in storage_schema_dict:
+                return False
+
+            # every type has to match
+            if storage_schema_dict[field.name].dataType != field.dataType:
+                return False
+
+        # All specified fields are found with the correct dataType.
+        # We can read the schema as-is (perhaps with a select statement
+        # to ensure column capitalization and order)
+        return True
+
+    def _tblproperties_alter_statements(self) -> List[str]:
+        """return that ALTER TABLE statements that will
+        transform the tblproperties of the base tabel into the target"""
+        statements = []
+
+        # treat tblproperties
+        if self.base.tblproperties != self.target.tblproperties:
```

use fail early pattern

---

done